main.go

```diff
@@ -58,50 +57,56 @@ func run(args []string) error {
 	case "new":
 		err := newCommand.Parse(args[2:])
 		if err != nil {
-			return err
+			return "", err
 		}
-		message, commandErr = command.NewCommand(ctx, *newProviderType, path, *newToken, *newGroup, *newApp, *newTag)
+		message, commandErr = withCopyOfWorkTree(newPath, func(workTreeCopy string) (string, error) {
+			return command.NewCommand(ctx, *newProviderType, workTreeCopy, *newToken, *newGroup, *newApp, *newTag)
+		})
 	case "promote":
 		err := promoteCommand.Parse(args[2:])
 		if err != nil {
-			return err
+			return "", err
 		}
-		message, commandErr = command.PromoteCommand(ctx, *promoteProviderType, path, *promoteToken)
+		message, commandErr = withCopyOfWorkTree(promotePath, func(workTreeCopy string) (string, error) {
+			return command.PromoteCommand(ctx, *promoteProviderType, workTreeCopy, *promoteToken)
+		})
 	case "status":
 		err := statusCommand.Parse(args[2:])
 		if err != nil {
-			return err
+			return "", err
 		}
-		message, commandErr = command.StatusCommand(ctx, *statusProviderType, path, *statusToken)
+		message, commandErr = withCopyOfWorkTree(statusPath, func(workTreeCopy string) (string, error) {
+			return command.StatusCommand(ctx, *statusProviderType, workTreeCopy, *statusToken)
+		})
 	default:
 		flag.PrintDefaults()
-		return fmt.Errorf("Unknown flag: %s", args[1])
+		return "", fmt.Errorf("Unknown flag: %s", args[1])
 	}
 
 	if commandErr != nil {
-		return commandErr
+		return "", commandErr
 	}
 
-	fmt.Println(message)
-
-	return err
+	return message, err
 }
 
-func setupFilesystem() (string, error) {
-	curPath, err := os.Getwd()
-	if err != nil {
-		return "", err
-	}
-
+func withCopyOfWorkTree(sourcePath *string, work func(string) (string, error)) (string, error) {
```

Comment:
Is `work` the best function name here? A personal preference is to make it clear that it's a function through, something like `workFn` - but may be my own weird preference.

Reply:
You're right. For example `taskFn` would be a better name. Fixed.